src/instructlab/model/backends/backends.py

```diff
@@ -219,31 +219,34 @@ def ensure_server(
         # pylint: enable=duplicate-code
     except ClientException:
         port = free_tcp_ipv4_port(host)
-        logger.debug("Using %i", port)
+        logger.debug(f"Using available port {port} for temporary model serving.")
 
         host_port = f"{host}:{port}"
         temp_api_base = get_api_base(host_port)
-        logger.debug(f"Starting a temporary server at {temp_api_base}...")
         llama_cpp_server_process = None
         vllm_server_process = None
 
         if backend == VLLM:
             # TODO: resolve how the hostname is getting passed around the class and this function
             vllm_server_process = server_process_func(port)
+            logger.info(f"Starting a temporary vLLM server at {temp_api_base}...")
             count = 0
             # TODO should this be configurable?
             vllm_startup_timeout = 300
             while count < vllm_startup_timeout:
-                sleep(1)
                 try:
                     list_models(
                         api_base=temp_api_base,
                         http_client=http_client,
                     )
-                    logger.debug(f"model at {temp_api_base} served on vLLM")
+                    logger.info(f"Successfully connected to vLLM at {temp_api_base}!")
                     break
                 except ClientException:
                     count += 1
+                    logger.info(
+                        f"Attempting to connect to starting vLLM server as {temp_api_base}, this might take a moment... {count}s/{vllm_startup_timeout}s"
```

Comment:
Does the list_models call return immediately the whole time vLLM is starting?

Reply:
it should fail a curl request until vllm is up

Reply:
> it should fail a curl request until vllm is up

Right, but does it wait at all or always fail fast.  I am asking because the timing will be off with only counting the sleep time if there is time spent waiting to connect.

Reply:
Ah I see what you're saying. It fails fast, but maybe it's better to not communicate the number of retried in terms of time.